core/templates/pages/exploration-player-page/services/command-executor.service.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A service that maintains a record of the state of the player,
+ *  like engine service.
+ */
+
+
+require('services/contextual/window-ref.service.ts');
+require(
+  'pages/exploration-player-page/services/window-wrapper-message.service');
+require('./window-wrapper-message.service');
+require(
+  'pages/exploration-player-page/services/current-interaction.service.ts');
+angular.module('oppia').factory('CommandExecutorService', [
+  'WindowWrapperMessageService',
+  function(WindowWrapperMessageService) {
+    var hostname = '';
+    var cachedOuterFrameMessage = '';
+    var initialized = false;
+    var currentInteraction = '';
+    var sendParentReadyState = function() {
+      WindowWrapperMessageService.postMessageToParent(
+        'Ready to receive hostname', '*');
+    };
+    var initialize = function() {
+      WindowWrapperMessageService.addEventListener(
+        'message', (event) => {
+          var messageArray = event.data.split(' ');
+          var command = messageArray[0];
+          var message = '';
+          for (var i = 1; i < messageArray.length; i++) {
+            message = message + messageArray[i] + ' ';
+          }
+          message = message.substr(0, message.length - 1);
+          if ((command !== 'HOSTNAME' && hostname === '')) {
+            return;
+          } else if (command === 'CONTINUE') {
+            commandToFunctionMap[command](currentInteraction);
+          } else if (command === 'HOSTNAME') {
+            hostname = message;
+            if (cachedOuterFrameMessage !== '' &&
+            cachedOuterFrameMessage !== undefined) {
+              WindowWrapperMessageService.postMessageToParent(
+                cachedOuterFrameMessage, hostname);
+            }
+          }
+        });
+      initialized = true;
+    };
+
+    var sendInteractionToOuterFrame = function(id) {
+      if (!initialized) {
+        return;
+      }
+      var stateToCommand = {
```

Comment:
interactionIdToCommand perhaps?

If you are using the commands elsewhere, make them named constants.

Also define this once outside instead, rather than doing so every time this function is called.